### DISCUSSION TITLE
[Sitemap v4.1.0] "output" is not allowed to be empty

### DISCUSSION THREAD

**Discussion by mwskwong** [gatsbyjs/gatsby#31381] [Help]
## Description
According to PR https://github.com/gatsbyjs/gatsby/pull/31130 and [Gatsby v3.5 release notes](https://www.gatsbyjs.com/docs/reference/release-notes/v3.5/#notable-bugfixes--improvements), it should now be possible to specify `output: ""` in `gatsby-sitemap-plugin@4.1.0`. Yet, it appears to reject an empty string value and still report the `"output" is not allowed to be empty` error, same as older versions.

### Steps to reproduce
1. Use any starter template
2. Install `gatsby-sitemap-plugin@4.1.0`
3. In `gatsby-config.js`, set the option `output` to `""`
i.e.
```js
{
  resolve: "gatsby-plugin-sitemap",
  options: {
    output: ""
  }
}
```

### Expected result
`gatsby-sitemap-plugin@4.1.0` should accept `output: ""`.

### Actual result
It rejected the value and an error is promoted instead.

### Environment
```

  System:
    OS: Windows 10 10.0.18363
    CPU: (12) x64 Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
  Binaries:
    Node: 15.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 7.11.1 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.9.4
  Browsers:
    Chrome: 90.0.4430.93
    Edge: Spartan (44.18362.449.0)
  npmPackages:
    gatsby: ^3.5.0 => 3.5.0 
    gatsby-plugin-emotion: ^6.5.0 => 6.5.0 
    gatsby-plugin-image: ^1.5.0 => 1.5.0 
    gatsby-plugin-manifest: ^3.5.0 => 3.5.0 
    gatsby-plugin-offline: ^4.5.0 => 4.5.0 
    gatsby-plugin-react-helmet: ^4.5.0 => 4.5.0 
    gatsby-plugin-robots-txt: ^1.6.2 => 1.6.2 
    gatsby-plugin-root-import: ^2.0.6 => 2.0.6 
    gatsby-plugin-sharp: ^3.5.0 => 3.5.0 
    gatsby-plugin-sitemap: ^4.1.0 => 4.1.0 
    gatsby-plugin-webpack-bundle-analyser-v2: ^1.1.22 => 1.1.21 
    gatsby-source-filesystem: ^3.5.0 => 3.5.0 
    gatsby-transformer-sharp: ^3.5.0 => 3.5.0 
  npmGlobalPackages:
    gatsby-cli: 3.0.0
```
```js
flags: {
  FAST_DEV: true,
  DEV_SSR: true,
  PRESERVE_FILE_DOWNLOAD_CACHE: true,
  PRESERVE_WEBPACK_CACHE: true,
  PARALLEL_SOURCING: true
}
```

**Comment by mwskwong** [comment]
### Workaround
Set `output` to `"/"` or `"."`.

**Comment by LekoArts** [comment]
Hi!

While @pedrolamas initially planned on allowing `" "` the PR (https://github.com/gatsbyjs/gatsby/pull/31130) diverged a bit and changed into making setting `"/"` actually work. So the goal of "Allows sitemap output on root of public folder" is achieved by that :)

**Reply by mwskwong** [reply]
So in other words, using `"/"` will no longer cause the double `/` in `<head>`?

**Reply by LekoArts** [reply]
Yep, correct
